src/Vex/Resource.h

```diff
@@ -1,10 +1,21 @@
 #pragma once
 
+#include <functional>
+#include <print>
+#include <span>
+#include <variant>
+
 #include <Vex/Handle.h>
+#include <Vex/RHIFwd.h>
 #include <Vex/Types.h>
 
+#include "RHI/RHIAllocator.h"
+
+#define VEX_USE_CUSTOM_ALLOCATOR_BUFFERS 1
```

Comment:
Nit: would change this to be wrapped by #ifndef / #endif to allow for potential user override.